api/v1alpha6/conversion.go

```diff
@@ -88,10 +88,36 @@ func restorev1alpha6MachineSpec(previous *OpenStackMachineSpec, dst *OpenStackMa
 	dst.Subnet = previous.Subnet
 }
 
+func restorev1alpha7MachineSpec(previous *infrav1.OpenStackMachineSpec, dst *infrav1.OpenStackMachineSpec) {
+	// PropagateUplinkStatus has been added in v1alpha7.
+	// We restore the whole Ports since they are anyway immutable.
+	dst.Ports = previous.Ports
+}
+
 func restorev1alpha7ClusterSpec(previous *infrav1.OpenStackClusterSpec, dst *infrav1.OpenStackClusterSpec) {
 	// APIServerLoadBalancer.Provider is new in v1alpha7
 	dst.APIServerLoadBalancer.Provider = previous.APIServerLoadBalancer.Provider
 	dst.Router = previous.Router
+	// PropagateUplinkStatus has been added in v1alpha7.
+	// We restore the whole Ports since they are anyway immutable.
+	if previous.Bastion != nil && previous.Bastion.Instance.Ports != nil {
```

Comment:
Do we have to check for `previous.Bastion.Instance != nil` here?

Reply:
That's ok, it's not a pointer.

It would be nice if the syntax was different though, wouldn't it? With some fields being pointers and others not and the syntax for dereferencing them being the same, having to remember to check this stuff manually is just a landmine even bigger than not having Optional types in the first place. Thanks, Go language designers 🤬